mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java

```diff
@@ -161,12 +163,45 @@ public void setRights(MailboxId mailboxId, MailboxACL mailboxACL, MailboxSession
 
     @Override
     public void setRights(MailboxPath mailboxPath, MailboxACL mailboxACL, MailboxSession session) throws MailboxException {
+        validateDomainsAreIdentical(mailboxPath.getUser(), mailboxACL);
         MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
         Mailbox mailbox = mapper.findMailboxByPath(mailboxPath);
 
         setRights(mailboxACL, mapper, mailbox);
     }
 
+    @VisibleForTesting
+    void validateDomainsAreIdentical(String user, MailboxACL mailboxACL) throws DifferentDomainException {
```

Comment:
the types are not aligned with the method name.
assertACLShareesBelongsToUserDomain ?
I don't really like the MailboxACL type much, I would rename it MailboxShares so that the method name would become assertSharesBelongsToUserDomain

Reply:
**MailboxShares** is not OK to me. You are adding a new term to our domain of knowledge. I don't remember we ever used the term **"shares"** at all will discussing delegation feature. Or if we did we presented "shared account". I'm really not fan of that naming, that:

 - Clashes with the entire content of the Sprint. Yes, I'm worried having to change hundreds of file import and rename logic manually.

I would rather implement it:
 - Extract the domain of the user *as an optional<String>*
 - If present I call a **assertAllEntryBelongToDomain** that IMO would not clash with the ACL naming.

Reply:
I'm ok to not rename MailboxACL, I was just discussing things.
However, ACL is about access control. It's orthogonal to mailbox types (Own, Delegated, Shared).
We should pay attention to not propagate MailboxACL in wrong contexts.